### PR TITLE
Bug fixes for billing

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Issue fix : #866 ,#1542, #1048

### CODE REVIEW

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -35,6 +35,8 @@ class Handler extends ExceptionHandler
      */
     public function report(Throwable $exception)
     {
+        // dd($exception);
```
Comment (sandesh556) [inline app/Exceptions/Handler.php:38]: Remove this comment and dd($exception) from the PR

File: storage/installed [hunk 2]
```diff
@@ -0,0 +1 @@
+Agora Invoicing Installer successfully INSTALLED on 2022/01/11 07:39:16am
```
Comment (sandesh556) [inline storage/installed:1]: Remove this also from the PR

**Review by sandesh556 — CHANGES_REQUESTED** [review]
Also check if those image files that are present in the PR are present in development or master. If not remove them also

File: app/Http/Controllers/Common/SettingsController.php [hunk 3]
```diff
@@ -209,6 +211,16 @@ public function postSettingsSystem(Setting $settings, Request $request)
         }
     }
 
+    public function changeLogo($id, $value)
```
Comment (sandesh556) [inline app/Http/Controllers/Common/SettingsController.php:214]: Add code comments to this fucntion

**Review by sandesh556 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Common/SettingsController.php [hunk 4]
```diff
@@ -209,6 +211,21 @@ public function postSettingsSystem(Setting $settings, Request $request)
...
+     * Get the id and value of the column.
+     *
+     * Remove the logo from the DB and local storage.
+     */
+    public function changeLogo($id, $value)
+    {
+        $log = Setting::find($id);
+        File::delete(public_path('common/images/').$log->$value);
+        File::delete(public_path('admin/images/').$log->$value);
+        $result = DB::table('settings')->where('id', $id)->update([$value => null]);
```
Comment (sandesh556) [inline app/Http/Controllers/Common/SettingsController.php:224]: use models instead of query builders

**Review by sandesh556 — CHANGES_REQUESTED** [review]

**Review by sandesh556 — APPROVED** [review]

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Last commit:-
image
Not working.
#866:- Not working.
image

#1542:- Not working.
image

#1048:- Not working.
b2

**Review by sandesh556 — APPROVED** [review]

**Review by priyadarshini2101 — CHANGES_REQUESTED** [review]
Tested the pr with existing db on server.

- [ ] Issue 1:Unable to delete the logo once uploaded in the system settings.

Before fix
jul13logo before fix

After fix
jul13logo after fix

Other Issues pointed are fixed.

Redirection Issue

Before fix

jul13redirect before fix

After fix

jul13redirect after fix


System Email should be unique

Before fix

jul13mail before fix

After fix

jul13mail after fix

**Review by RafficMohammed — APPROVED** [review]

**Comment by sowmiyasubramaniyan** [comment]
**Issue 1:Unable to delete the logo once uploaded in the system settings**
  Fixed

**Review by priyadarshini2101 — CHANGES_REQUESTED** [review]
Tested the pr with existing db on server.

Issue 1:When the file uploaded is of type txt or someother unsupported format for either of one Admin Panel logo/Favicon/Client Panel logo . The alert message is displaying. But,the files uploaded to these are all getting vanished. —--Is it fine?

Issue 2:When the logo is deleted for either of one Admin Panel logo/Favicon/Client Panel logo, the API is not showing.

jul20logo

jul20logo


Issue 3:When same logo is uploaded for Admin Panel logo/Favicon/Client Panel logo, and try to delete the logo of Admin Panel, the same logo uploaded in other places the images are breaking.

Note 1:Popup alert message for Deletion of logo is not user friendly.
Note 2:Alert messages are not vanishing after displaying staying for a long.

**Review by priyadarshini2101 — APPROVED** [review]
Created seperate issue and approving this pr.

https://github.com/ladybirdweb/agora-invoicing-community/issues/1875

**Review by Sheetal34 — APPROVED** [review]
Tested for mentioned fixes. Working fine

**Pull request merged**